### PR TITLE
Audio augmentation

### PR DESCRIPTION
**Pull request by pietraldo**
- created script for data augmentation (noise, pitch, speed, echo, volume, reverb)
 script gets in args path to file and output file path and give each effect and save this with this effect. You can also mix those effects

- corrected get random file function

Example changes after noise effect in spectrograms:
a_echo0
a_noise0
a_original0
a_pitch0
a_reverb0
a_speed0
a_volume0

### CODE REVIEW

**Comment by kryczkal** [comment]
LGTM, except lacks tests to see if the functions work properly.

File: src/scripts/audio_augmentation.py [hunk 1]
```diff
@@ -0,0 +1,237 @@
...
+from src.pipeline.audio_data import AudioData
+
+
+DEFAULT_SEMITONES = 6
+DEFAULT_SPEED_FACTOR = 0.5
+DEFAULT_NOISE_LEVEL = 0.03
+DEFAULT_GAIN_DB = -10
+DEFAULT_REVERB_AMOUNT = 0.1
+DEFAULT_ECHO_DELAY = 0.25
+DEFAULT_ECHO_DECAY = 0.6
```
Comment (Jlisowskyy) [inline src/scripts/audio_augmentation.py]: Please move the values to the constants.py file

File: src/scripts/audio_augmentation.py [hunk 2]
```diff
@@ -0,0 +1,237 @@
...
+
+def save_file(audio_data: AudioData, file_path: str) -> None:
+    """
+    Saves the given audio data to the specified file path.
+
+    Args:
+        audio_data (AudioData): The audio data to save.
+        file_path (str): The path where the audio file should be saved.
+    """
+    sf.write(file_path, audio_data.audio_signal, audio_data.sample_rate)
```
Comment (Jlisowskyy) [inline src/scripts/audio_augmentation.py]: one line function wrappers?

File: src/scripts/audio_augmentation.py [hunk 3]
```diff
@@ -0,0 +1,237 @@
...
+    Returns:
+        AudioData: The audio data with echo.
+    """
+    delay_samples = int(audio_data.sample_rate * delay)
+    echo_signal = np.zeros(len(audio_data.audio_signal) + delay_samples)
+    echo_signal[: len(audio_data.audio_signal)] = audio_data.audio_signal
+    for i, sample in enumerate(audio_data.audio_signal):
+        echo_signal[i + delay_samples] += sample * decay
+    echo_signal = echo_signal[: len(audio_data.audio_signal)]
+    return AudioData(echo_signal, audio_data.sample_rate)
```
Comment (Jlisowskyy) [inline src/scripts/audio_augmentation.py:131]: Functions looks good, but for sure they should not be implemented inside scripts directory.

File: src/scripts/audio_augmentation.py [hunk 4]
```diff
@@ -0,0 +1,237 @@
...
+    """
+    parser = argparse.ArgumentParser(
+        description="Generates and saves augmented versions of an audio file.")
+    parser.add_argument(
+        "--file_path", '-f', type=str, required=True, help="Path to the input audio file.")
+    parser.add_argument(
+        "--output_path", '-o', type=str, required=True, help="Path to save the output audio file.")
+
+    args = parser.parse_args(argv)
+    process(args.file_path, args.output_path)
```
Comment (Jlisowskyy) [inline src/scripts/audio_augmentation.py]: This part may stay here or be moved to the tests, as we do not expect this functionality to serve as a ascript. Instead, its main purpose is to provide a function that can be used within the prepare_database script to create multiple .npy files without duplicating audio files, saving a significant amount of time.

Comment (Jlisowskyy) [reply src/scripts/audio_augmentation.py]: You should also hook thus function up into the preapre_databse script

**Review by Jlisowskyy — CHANGES_REQUESTED** [review]
Small changes needed

**Comment by Jlisowskyy** [comment]
Please also attach some samples how it works to the PR description, so anyone revewing the code can see also the results

File: src/scripts/audio_augmentation.py [hunk 5]
```diff
@@ -0,0 +1,215 @@
...
+    Args:
+        audio_data (AudioData): The original audio data.
+        noise_level (float): The standard deviation of the noise to add.
+
+    Returns:
+        AudioData: The audio data with added noise.
+    """
+    noise = np.random.normal(0, noise_level, audio_data.audio_signal.shape)
+    noisy_signal = audio_data.audio_signal + noise
+    return AudioData(noisy_signal, audio_data.sample_rate)
```
Comment (al-gor1thm) [inline src/scripts/audio_augmentation.py:73]: We can dump this as @KwiatkowskiML already made a similar function

**Review by KwiatkowskiML — APPROVED** [review]
lgtm

File: src/scripts/audio_augmentation.py [hunk 6]
```diff
@@ -0,0 +1,215 @@
...
+    if 'volume' in options:
+        audio_data = change_volume(audio_data, gain_db)
+
+    if 'reverb' in options:
+        audio_data = add_reverb(audio_data, reverb_amount)
+
+    if 'echo' in options:
+        audio_data = add_echo(audio_data, echo_delay, echo_decay)
+
+    return audio_data
```
Comment (al-gor1thm) [inline src/scripts/audio_augmentation.py:178]: Some random seeded selector would be nice (TODO?)

**Review by al-gor1thm — APPROVED** [review]

**Review by Jlisowskyy — APPROVED** [review]
Ehh, i hope some random selection was done

File: src/scripts/audio_augmentation.py (hunk 6 above)
Comment (al-gor1thm) [reply src/scripts/audio_augmentation.py:178]: alright me or Michał'll do it

**Comment by al-gor1thm** [comment]
> Ehh, i hope some random selection was done

We'll do it in next PR

**Pull request merged**